programs/drift/src/math/margin.rs

```diff
@@ -783,21 +783,21 @@ pub fn calculate_max_withdrawable_amount(
             spot_market_map,
             oracle_map,
             None,
-            false,
+            true,
         )?;
 
     let spot_market = &mut spot_market_map.get_ref(&market_index)?;
 
-    if spot_market.initial_asset_weight == 0 {
```

Comment:
guess i was thinking, even if the weight was 0, it could still be swapped for an asset with a weight, so allowing for withdraw without checks decreases chance user account would be in good standing

Reply:
i think it will throw a math error below if this return early isn't there